p2p/src/network/pubsub/p2p_network_pubsub_reducer.rs

```diff
@@ -49,28 +70,70 @@ impl P2pNetworkPubsubState {
                 protocol,
             } => {
-                let state =
-                    self.clients
-                        .entry(*peer_id)
-                        .or_insert_with(|| P2pNetworkPubsubClientState {
-                            protocol: *protocol,
-                            addr: *addr,
-                            outgoing_stream_id: Some(*stream_id),
-                            message: pb::Rpc {
-                                subscriptions: vec![],
-                                publish: vec![],
-                                control: None,
-                            },
-                            buffer: vec![],
-                            incoming_messages: vec![],
-                        });
+                let state = pubsub_state.clients.entry(*peer_id).or_insert_with(|| {
+                    P2pNetworkPubsubClientState {
+                        protocol: *protocol,
+                        addr: *addr,
+                        outgoing_stream_id: Some(*stream_id),
+                        message: pb::Rpc {
+                            subscriptions: vec![],
+                            publish: vec![],
+                            control: None,
+                        },
+                        buffer: vec![],
+                        incoming_messages: vec![],
+                    }
+                });
                 state.outgoing_stream_id = Some(*stream_id);
                 state.protocol = *protocol;
                 state.addr = *addr;
+
+                let (dispatcher, state) = state_context.into_dispatcher_and_state();
+                let config: &P2pConfig = state.substate()?;
+                let state: &P2pNetworkPubsubState = state.substate()?;
+
+                let subscrption = {
+                    let msg = pb::Rpc {
+                        subscriptions: vec![pb::rpc::SubOpts {
+                            subscribe: Some(true),
+                            topic_id: Some(TOPIC.to_owned()),
+                        }],
+                        publish: vec![],
+                        control: None,
+                    };
+                    Some(P2pNetworkPubsubAction::OutgoingMessage {
+                        msg,
+                        peer_id: *peer_id,
+                    })
+                };
+                let Some(map) = state.topics.get(TOPIC) else {
+                    // must have this topic already
+                    return Ok(());
+                };
+                let mesh_size = map.values().filter(|s| s.on_mesh()).count();
+                let graft = if mesh_size < config.meshsub.outbound_degree_desired {
+                    Some(P2pNetworkPubsubAction::Graft {
+                        peer_id: *peer_id,
+                        topic_id: TOPIC.to_owned(),
+                    })
+                } else {
+                    None
+                };
+                for action in subscrption.into_iter().chain(graft) {
```

Comment:
Now that we can enqueue actions do we really need this loop? Why not just put the `dispatcher.push` inline instead of returning `Some(action)`?